src/main/java/edu/cornell/kfs/pmw/batch/service/impl/PaymentWorksTaxRuleDependencyServiceImpl.java

```diff
@@ -0,0 +1,261 @@
+    }
+    
+    private void populateW8Attributes(KfsVendorDataWrapper kfsVendorDataWrapper, PaymentWorksVendor pmwVendor, TaxRule taxRule) {
+        if (StringUtils.isNotBlank(pmwVendor.getW8SignedDate())) {
+            LOG.debug("populateW8Attributes, setting W8 values");
+            kfsVendorDataWrapper.getVendorDetail().getVendorHeader().setVendorW8BenReceivedIndicator(true);
+            kfsVendorDataWrapper.getVendorDetail().getVendorHeader().setVendorW8SignedDate(buildDateFromString(pmwVendor.getW8SignedDate()));
+            kfsVendorDataWrapper.getVendorDetail().getVendorHeader().setVendorW8TypeCode(taxRule.w8TypeCode);
+            paymentWorksBatchUtilityService.createNoteRecordingAnyErrors(kfsVendorDataWrapper, 
+                    configurationService.getPropertyValueAsString(PaymentWorksKeyConstants.NEW_VENDOR_PVEN_NOTES_W8_URL_EXISTS_MESSAGE), 
+                    PaymentWorksConstants.ErrorDescriptorForBadKfsNote.W8.getNoteDescriptionString());
+        } else {
+            LOG.debug("populateW8Attributes, NOT setting W8 values");
+            kfsVendorDataWrapper.getVendorDetail().getVendorHeader().setVendorW8BenReceivedIndicator(false);
+        }
+    }
+    
+    protected Date buildDateFromString(String  dateString) {
+        Date w8Date = new Date(getDateTimeFormatter().parseDateTime(dateString).getMillis());
+        LOG.info("buildDateFromString, w8Date: " + w8Date);
+        return w8Date;
+    }
+    
+    private void populateFirstLastLegalName(PaymentWorksVendor pmwVendor, KfsVendorDataWrapper kfsVendorDataWrapper) {
+        kfsVendorDataWrapper.getVendorDetail().setVendorFirstLastNameIndicator(true);
+        String vendorLastName = truncateValueToMaxLength(pmwVendor.getRequestingCompanyLegalLastName(), VendorConstants.MAX_VENDOR_NAME_LENGTH);
+        kfsVendorDataWrapper.getVendorDetail().setVendorLastName(vendorLastName);
+        kfsVendorDataWrapper.getVendorDetail().setVendorFirstName(truncateLegalFirstNameToMaximumAllowedLengthWhenFormattedWithLegalLastName(vendorLastName, pmwVendor.getRequestingCompanyLegalFirstName(), VendorConstants.NAME_DELIM, VendorConstants.MAX_VENDOR_NAME_LENGTH));
+    }
+    
+    private KfsVendorDataWrapper populateBusinessLegalName(PaymentWorksVendor pmwVendor, KfsVendorDataWrapper kfsVendorDataWrapper) {
+        kfsVendorDataWrapper.getVendorDetail().setVendorName(truncateValueToMaxLength(pmwVendor.getRequestingCompanyLegalName(), VendorConstants.MAX_VENDOR_NAME_LENGTH));
+        kfsVendorDataWrapper.getVendorDetail().setVendorFirstLastNameIndicator(false);
+        kfsVendorDataWrapper.getVendorDetail().setVendorFirstName(KFSConstants.EMPTY_STRING);
+        kfsVendorDataWrapper.getVendorDetail().setVendorLastName(KFSConstants.EMPTY_STRING);
+        return kfsVendorDataWrapper;
+    }
+    
+    protected static String truncateValueToMaxLength(String inputValue, int maxLength) {
```

Comment:
Could you use StringUtils.truncate instead?

Reply:
yes